### ISSUE TITLE
User should be able to start a thread in a message

### ISSUE THREAD

**Issue by tim-habitat** [TryQuiet/quiet#1817]
It would be great if we could reply / start a thread from a message!

Couldn't find a related issue, but i am sure there is one.

**Comment by leblowl** [comment]
Thanks for the issue! I personally like threads. Do we have any plans for a feature like this @holmesworcester?

**Comment by holmesworcester** [comment]
Yes definitely. We'll start by working with our designer to decide how this will work. I'm eager to get your feedback!

What model of threads do each of you prefer? Slack? Telegram? ...Zulip?

**Comment by leblowl** [comment]
I think I really only have experience with Slack, which I've generally enjoyed. I can play around with other implementations and see where they differ.

**Comment by holmesworcester** [comment]
Some more notes from users on what we're trying to achieve:

Threads are an important feature for users to have busy private channels, because it lets you separate a conversation without giving it its own topic or re-choosing the invitees. 

It also lets people participate in conversations without feeling like they are interrupting or demanding too much attention. 

We've thought about Facebook-style threads where every post has a section for replies, but it seems that this is not what people want: apparently Google Chat used to have the option to create facebook style posts (which would have a space to reply in a thread) but nobody did it, so recent versions of Google Chat do threads the way Slack does. 

We've generally seen that Slack is the thing people say when you ask them what thread system they like the best. The person replying (not the person sending the initial message) decides to create the thread, and Slack also does a good job of showing you new messages in old threads, which is important too. 

The only thing Slack doesn't do that well (according to some users) is encourage people to reply in threads, especially on mobile. Users who are less comfortable with the tool might mean to reply in a thread but reply in the main channel. I'm not sure how to address this beyond, perhaps, having a quick-reply-from-notification feature that defaults to replying in a thread. We should see if there are any other widely used options for this.

**Comment by holmesworcester** [comment]
[Loom](https://www.loom.com/share/a690115728424f3a9b80caf945cbeb05) and [Figma](https://www.figma.com/file/ri4AfHBYsrVsP4WRz3P6z4/Threads?type=design&node-id=13-37&mode=design&t=LzLlSGGDW4A5PZt2-4)

**Comment by jgaylor** [comment]
@holmesworcester here's a more recent draft I worked up this morning based on [your comment here to take another look at threads](https://github.com/TryQuiet/quiet/issues/2031#issuecomment-2015875337). 

[Loom](https://www.loom.com/share/f643c4990eb2422e8aa34ca566443f50) and [Figma](https://www.figma.com/file/ri4AfHBYsrVsP4WRz3P6z4/Threads?type=design&node-id=876%3A10658&mode=design&t=b24dBZ9UacMXERqO-1)

**Comment by holmesworcester** [comment]
TODO: show a message that starts a thread and has reactions

**Comment by jgaylor** [comment]
@holmesworcester had some other things that were missing here too, so I've made a new [Figma](https://www.figma.com/file/ri4AfHBYsrVsP4WRz3P6z4/Threads?type=design&node-id=880%3A16577&mode=design&t=b24dBZ9UacMXERqO-1) and [Loom](https://www.loom.com/share/b08fc78195144ee396d0ceaba01e98a8) with updates

Main things:
1. Show messages with reactions
2. Show message and user flows for messages that are already threads, and messages that aren't yet a thread. (This differentiation was missing before)

**Comment by jgaylor** [comment]
Updated [Figma](https://www.figma.com/file/ri4AfHBYsrVsP4WRz3P6z4/Threads?type=design&node-id=887%3A25190&mode=design&t=aUkrPUvUktS9Zmu0-1)

**Comment by holmesworcester** [comment]
User feedback: threads shouldn't cover the parent channel when side by side. Let's update that in the designs.

**Comment by holmesworcester** [comment]
User requested threads be accessible alongside the main view (if the window is big enough) and not have threads occluding the main view.

What happens if someone clicks another user's profile inside a thread? Can they navigate back to the thread?

What about when they open settings or their own user profile.

**Comment by jgaylor** [comment]
Threads Draft 6 [Figma](https://www.figma.com/design/ri4AfHBYsrVsP4WRz3P6z4/Threads?node-id=986-7686&t=0kuQ92X4aiYSyWaH-4) and [Loom](https://www.loom.com/share/b19f9c526fc44268a369cb6057754795)

Updates:
1. Entire spec cleanup: many things were outdated, various misc reconciling
2. Updated to use a flexible panel for threads
3. Window panel scaling general guidelines and mockups
4. Navigating the panel general guidelines and mockups

**Comment by holmesworcester** [comment]
We might want to move all the "side panel" mechanics to its own library so that it's not just in the threads view, so that any frontend contributor (e.g. mike) has a clear path. 

For example, when we're showing settings or sheets elsewhere in other mockups we should use the flex panel.

If we need something to refer to in the meantime, this is the key section: https://www.figma.com/design/ri4AfHBYsrVsP4WRz3P6z4/Threads?node-id=988-44882&t=me5qsoxNspsG0gYX-4

I should review this.
